src/server.rs

```diff
@@ -255,27 +256,40 @@ impl rpc::rusk_server::Rusk for Rusk {
 
         // Make output note
         let pk: PublicKey = pk.try_into()?;
-        let (note, blinding_factor) =
-            TransparentNote::output(&pk, request.value);
+        let output: TransactionOutput = match request.obfuscated {
```

Comment:
You might not need the match, is a check against just a boolean after all. You should be fine with:
```
let output = if request.obfuscated {
    let (note, blinding_factor) =
        ObfuscatedNote::output(&pk, request.value);
        note.to_transaction_output(request.value, blinding_factor, pk)
} else {
    let (note, blinding_factor) =
        TransparentNote::output(&pk, request.value);
    note.to_transaction_output(request.value, blinding_factor, pk)
}
```
The same should be applied for the rest of the code.
(I also assume rust should be able to infer the type of `output`)